_posts/2024-11-08-how-ai-tools-sabotage-juniors-learning-curve.md

```diff
@@ -0,0 +1,192 @@
+comments: true
+---
+
+> You could literally hammer a nail into a shoe, and it would get the job done. But does it really?
+
+{:.no_toc}
+- TOC
+{:toc}
+
+## Introduction
+
+Using AI as a junior developer often provides a sense of hope and easy access to information, 
+but this convenience comes with its own set of challenges when real (client) problems arise. 
+In the spirit of preserving knowledge and fostering honest communication, 
+I felt compelled to share my perspective on this journey.
+
+This article stems from a moment of reflection during a drive home,
+where I pondered my apprenticeship and the learning process as a junior developer. 
+This is aimed to comfort fellow juniors and newcomers and to offer relevant feedback to colleagues.
+
+<img alt="junior_reading" src="{{ '/img/2024-11-08-how-ai-sabotages-juniors-learning-curve/baby-reading.png' | prepend: site.baseurl }}" style="float: right; max-width: 180px;">
+
+## Overview: The Dual Nature of AI in the Junior Developer Experience
+
+As junior developers, we are heavy consumers of information. 
+We love that "cooked" knowledge provided by experts and often accept it unquestioningly, building on it.
+The depth of these insights and experiences helps us retain information, carving it into stone.
+
+In contrast, the quick, easy information provided by AI tools may feel useful at the moment but is often fleeting, 
+like writing on the beach—easily erased by the next wave and quickly forgotten.
+I've experienced the feeling of solving a problem with AI’s help only to realize later that I can't remember how or why it worked.
+This led me to reflect on the types of information that stick and those that don’t.
+
+### When AI is Incredibly Beneficial
+
+AI can be highly advantageous when:
+
+- **We need help with technologies outside our primary focus.**  
+  For example, when I needed front-end code despite my primary focus being back-end development, AI tools provided fast and reliable assistance.
+  This helps junior developers fill in knowledge gaps quickly, enabling them to move forward without getting stuck.
+   For instance, I was not familiar with Angular, but using AI to prompt me on how components work made it much easier to refactor and update my code.
+
+- **AI provides code snippets, templates, or solutions for unfamiliar areas.**  
+  AI can provide code snippets, templates, and solutions for areas that may be unfamiliar to developers. 
+  This capability can significantly boost confidence and momentum, 
+  enabling users to progress through projects more efficiently.
+  For junior developers, this support can be particularly valuable when they encounter obstacles, 
+  allowing them to overcome challenges and continue their work. 
+ 
+
+
+
+### The Drawbacks of Over-Reliance on AI
+
+Despite the numerous benefits that AI can offer, 
+over-dependence on these tools can lead to significant drawbacks that affect our long-term development and problem-solving capabilities.
```

Comment:
een titel voor dit beetje. Dat is wel heel weinig. Misschien dan beter weglaten?